plugins/modules/azure_rm_virtualmachine.py

```diff
@@ -86,6 +86,28 @@
             - A valid Azure VM size value. For example, C(Standard_D4).
             - Choices vary depending on the subscription and location. Check your subscription for available choices.
             - Required when creating a VM.
+    priority:
+        description:
+            - Priority of the VM.
+        choices:
+            - Regular
+            - Spot
+        default: Regular
+    eviction_policy:
+        description:
+            - Specifies the eviction policy for the Azure Spot virtual machine.
+            - Requires priority to be set to Spot.
+        choices:
+            - Deallocate
+            - Delete
+        default: Deallocate
```

Comment:
Argument_spec defines default as (None) but documentation defines default as ('Deallocate')!

Reply:
The actual behavior as processed by Azure Python SDK is actually a little bit more complex.

By default if Priority is set to None, this field will be ignored. If Priority is set to "Spot", the default value will be Deallocate (even if payload sent is None).

Would ```default: None ('Deallocate' if priority: Spot) ``` would be suitable for the help section?

Reply:
```suggestion

```